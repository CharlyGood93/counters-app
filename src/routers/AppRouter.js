import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';
import React from 'react'
import { WelcomeScreen } from '../components/welcome/WelcomeScreen';
import { MainScreen } from '../components/main/MainScreen';

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <Switch>
                    <Route exact path="/" component={WelcomeScreen} />
                    <Route exact path="/counters" component={MainScreen} />
                    <Redirect to="/" />
                </Switch>
            </Router>
        </div>
    )
}
