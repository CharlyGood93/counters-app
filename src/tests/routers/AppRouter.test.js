import React from 'react';
import { shallow } from "enzyme";
import { AppRouter } from "../../routers/AppRouter";

describe('Testing <AppRouter />', () => {
    test('Should show the <AppRouter /> component correctly', () => {
        const wrapper = shallow(<AppRouter />);
        expect(wrapper).toMatchSnapshot();
    });
});
