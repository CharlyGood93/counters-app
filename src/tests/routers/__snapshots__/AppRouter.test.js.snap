// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AppRouter /> Should show the <AppRouter /> component correctly 1`] = `
<div>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/counters"
      />
      <Redirect
        to="/"
      />
    </Switch>
  </BrowserRouter>
</div>
`;
